pkg/api/controller.go

```diff
@@ -2225,6 +2225,14 @@ func (c *Controller) handleAPIErrorCallback(ctx context.Context, w http.Response
 
 	log := c.Logger.WithContext(ctx).WithError(err)
 
+	// Handle Hook Errors
+	var hookAbortErr *graveler.HookAbortError
+	if errors.As(err, &hookAbortErr) {
+		log.WithField("run_id", hookAbortErr.RunID).Warn("aborted by hooks")
+		cb(w, r, http.StatusPreconditionFailed, err)
+		return true
+	}
```

Comment:
The HookAbortErr can hold (wrapped by graveler.ErrPreconditionFailed), so the default handler implementation will return `http.StatusPreconditionFailed`.
All is left is the logging which I think we should keep in all places we trigger hooks and not in the central location.

My two concerns are:
- logging related to the operation, today we capture run_id tomorrow we will need the full content (ex: source branch and etc)
- the downside in having central error location is we can't map where this error is being used.

Reply:
1. HookAbortErr already wraps the underlying error from the hook execution.
2. Logging is coupled with checking that we actually had a HookAbortErr which brings us back to exactly the same situation where we were - missing handling because we forgot to add the same code block in another operation that uses hooks